File: src/2023/day-02/python/part_2.py
from functools import reduce

cube_num = {"red": 12, "green": 13, "blue": 14}

def Part2(input):
    data = tuple(ln.strip() for ln in input.strip().split("\n"))
    m = 0
    for line in data:
        min_cube = {"red": 0, "green": 0, "blue": 0}
        rest = line.split(":")
        rest.pop(0)[5:]
        for g in [g.split(",") for g in rest[0].split(";")]:
            for v, k in [c.split() for c in g]:
                min_cube[k] = max(min_cube[k], int(v))
        m += reduce(lambda x, y: x * y, min_cube.values())
    return m

File: src/2022/day-02/typescript/part-2.ts
// advent of code 2022 day 2 part 2

export function Part2(input: string) {
  const data = input.trim().split(/\n/).map((line) => line.split(/\s/));
  return data.map((p) => p[0]).map((p1, index) => {
    const p2 = data.map((p) => p[1])[index];
    if (p2 == 'Z') {
      if (p1 == 'A') return 2 + 6;
      else if (p1 == 'B') return 3 + 6;
      else return 1 + 6;
    } else if (p2 == 'Y') {
      if (p1 == 'A') return 1 + 3;
      else if (p1 == 'B') return 2 + 3;
      else return 3 + 3;
    } else {
      if (p1 == 'A') return 3 + 0;
      else if (p1 == 'B') return 1 + 0;
      else return 2 + 0;
    }
  }).reduce((a, b) => a + b);
}
